--- a/lib/Doctrine/ORM/UnitOfWork.php	2017-12-16 18:57:51.000000000 -0800
+++ b/lib/Doctrine/ORM/UnitOfWork.php	2023-08-29 17:59:42.000000000 -0700
@@ -2633,7 +2633,7 @@
                         $class->reflFields[$field]->setValue($entity, $data[$field]);
                         $this->originalEntityData[$oid][$field] = $data[$field];
 
-                        continue;
+                        break;
                     }
 
                     $associatedId = array();
@@ -2662,7 +2662,7 @@
                         $class->reflFields[$field]->setValue($entity, null);
                         $this->originalEntityData[$oid][$field] = null;
 
-                        continue;
+                        break;
                     }
 
                     if ( ! isset($hints['fetchMode'][$class->name][$field])) {
